src/utils/auth.ts

```diff
@@ -0,0 +1,19 @@
+import express from 'express';
+const jwt = require('jsonwebtoken');
+
+// TODO: Test out this middleware once login endpoint is done and returns a JSON web token
+const authenticateJWT = async (req: any, next: any) => {
+    const authHeader = req.headers.authorization;
+
+    if (authHeader) {
+        const token = authHeader.split(' ')[1];
+        try {
+            const user = await jwt.verify(token, process.env.ACCESS_TOKEN_SECRET)
+            req.user = user.id
```

Comment:
Would be best to store the user, instead of just the ID - that way we can also ensure that there is a valid user with that ID. It also means we can check other fields of the user if needed

```suggestion
            req.user = await prisma.user.findUniqueOrThrow({where:{id: user.id}})
```